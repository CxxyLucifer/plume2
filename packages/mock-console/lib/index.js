"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var MockConsole = /** @class */ (function () {
    function MockConsole() {
        this._logs = [];
        this.mock();
    }
    MockConsole.prototype.mock = function () { };
    MockConsole.prototype.mockFn = function () {
        var _this = this;
        return jest.fn(function (log) { return _this._logs.push(log); });
    };
    Object.defineProperty(MockConsole.prototype, "logs", {
        get: function () {
            return this._logs;
        },
        enumerable: true,
        configurable: true
    });
    MockConsole.prototype.reset = function () {
        this._logs = [];
    };
    return MockConsole;
}());
exports.MockConsole = MockConsole;
var MockWarn = /** @class */ (function (_super) {
    __extends(MockWarn, _super);
    function MockWarn() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockWarn.prototype.mock = function () {
        console.warn = this.mockFn();
    };
    return MockWarn;
}(MockConsole));
exports.MockWarn = MockWarn;
var MockLog = /** @class */ (function (_super) {
    __extends(MockLog, _super);
    function MockLog() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockLog.prototype.mock = function () {
        console.log = this.mockFn();
    };
    return MockLog;
}(MockConsole));
exports.MockLog = MockLog;
