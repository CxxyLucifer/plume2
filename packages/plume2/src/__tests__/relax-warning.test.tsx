import { MockWarn } from 'mock-console';
import React from 'react';
import renderer from 'react-test-renderer';
import { Actor, Relax, Store, StoreProvider } from '..';

class HelloActor extends Actor {
  defaultState() {
    return { id: 1 };
  }
}

class AppStore extends Store {
  bindActor() {
    return [HelloActor];
  }
}

@StoreProvider(AppStore, { debug: true })
class Home extends React.Component {
  render() {
    let List = [];
    for (let i = 0; i < 20; i++) {
      List.push(<Test key={i} />);
    }
    return <div>{List}</div>;
  }
}

@Relax
class Test extends React.Component {
  static relaxProps = {};
  render() {
    return <div>hello</div>;
  }
}

it('test output include warning', () => {
  const mock = new MockWarn();
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(mock.logs).toEqual([
    'you have to many Relax(Test) component, May be effect performance!'
  ]);
});
