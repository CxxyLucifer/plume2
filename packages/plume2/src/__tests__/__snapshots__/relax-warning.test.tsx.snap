// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test output include warning 1`] = `
<div>
  <div>
    hello
  </div>
  <div>
    hello
  </div>
  <div>
    hello
  </div>
  <div>
    hello
  </div>
  <div>
    hello
  </div>
  <div>
    hello
  </div>
  <div>
    hello
  </div>
  <div>
    hello
  </div>
  <div>
    hello
  </div>
  <div>
    hello
  </div>
  <div>
    hello
  </div>
  <div>
    hello
  </div>
  <div>
    hello
  </div>
  <div>
    hello
  </div>
  <div>
    hello
  </div>
  <div>
    hello
  </div>
  <div>
    hello
  </div>
  <div>
    hello
  </div>
  <div>
    hello
  </div>
  <div>
    hello
  </div>
</div>
`;

exports[`test output include warning 2`] = `
Array [
  "%cplume2@1.1.0ðŸš€",
  "StoreProvider(Home) will mount ðŸš€",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
  "Relax(Test) willMount rx store false ðŸš€",
  "props => {
  \\"relaxProps\\": {}
}",
]
`;
